### PR TITLE
xm88 cursor

### PR DESCRIPTION
**Pull request by harryob**
it's finally here

:cl: thwomper, harryob
ui: your mouse cursor is now special when using the xm-88
/:cl:

https://user-images.githubusercontent.com/55142896/212126418-e6262028-6a5d-4b84-9011-f3baf6a60457.mp4

the animation also fills up with the penetration level or w/e

### CODE REVIEW

File: code/modules/projectiles/guns/lever_action.dm [hunk 1]
```diff
@@ -412,6 +412,58 @@ their unique feature is that a direct hit will buff your damage and firerate
...
+/obj/item/weapon/gun/lever_action/xm88/proc/update_fired_mouse_pointer(mob/user)
+	SIGNAL_HANDLER
+
+	user.client?.mouse_pointer_icon = get_fired_mouse_pointer(floating_penetration)
+	addtimer(CALLBACK(src, PROC_REF(update_mouse_pointer), user), 0.4 SECONDS, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_CLIENT_TIME)
+
+/obj/item/weapon/gun/lever_action/xm88/proc/update_mouse_pointer(mob/user)
+	user.client?.mouse_pointer_icon = get_mouse_pointer(floating_penetration)
+
+/obj/item/weapon/gun/lever_action/xm88/proc/get_mouse_pointer(level)
```
Comment (Zonespace27) [inline code/modules/projectiles/guns/lever_action.dm:436]: can't both these procs be converted to `return 'icons/effects/mouse_pointer/xm88/xm88-[level].dmi'`?

Comment (harryob) [reply code/modules/projectiles/guns/lever_action.dm:436]: god i wish - ' ' can't have expressions in them, and file("icons/whatever/xm88-[level]") also... does not work. mouse_pointer_icon is a truly cursed

Comment (Zonespace27) [reply code/modules/projectiles/guns/lever_action.dm:436]: > god i wish - ' ' can't have expressions in them, and file("icons/whatever/xm88-[level]") also... does not work. mouse_pointer_icon is a truly cursed var

How about `icon("icons/whatever/xm88-[level]")`?

Comment (harryob) [reply code/modules/projectiles/guns/lever_action.dm:436]: this also doesn't work 😭

**Review by FransFeiffer — APPROVED** [review]
Sprites-wise, this is great, LGTM

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**